src/main/java/imsam/Main.java

```diff
@@ -20,25 +25,20 @@ public class Main extends Command {
 
     @Argument(required=true,index=0,metaVar="command",usage="subcommands, e.g., {generate|simulate}",handler=SubCommandHandler.class)
     @SubCommands({
-        @SubCommand(name="generate",impl=SparseModelGenerator.class),
+        @SubCommand(name="generate",impl=MGenCommand.class),
+        @SubCommand(name="mgen",impl=MGenCommand.class),
         @SubCommand(name="simulate",impl=ScaffoldImportanceSampling.class),
         @SubCommand(name="sim",impl=ScaffoldImportanceSampling.class),
     })
-    protected Command command; // Use Callable instead of Runnable to allow exceptions
+    protected Command command;
 
 
     public static void main(String[] args) {
         Main main = new Main();     // This looks weird, but an instance of the class is required by args4j
         CmdLineParser parser = new CmdLineParser(main);
         try {
             parser.parseArgument(args);
-            if (main.verboseTrace) {
-                setLogLevel(Level.TRACE);
-            } else if (main.verboseDebug) {
-                setLogLevel(Level.DEBUG);
-            } else if (main.verboseInfo) {
-                setLogLevel(Level.INFO);
-            }
+            main.readLoggingArgs();
```

Comment:
Isn't this called in `main.command.entryPoint()`?

Reply:
It needs to be called in both places. One to catch `imsam -v <command>` and the other for `imsam <command> -v`. The second overrides the first so `imsam -vvv <command> -v` would result in level one verbose logging.